covid_model/fit_disconnected_co_regions.py

```diff
@@ -0,0 +1,92 @@
+    parser.add_argument("-lb", "--look_back", type=int, help="the number of (default 14-day) windows to look back and refit; defaults to refitting all windows")
+    parser.add_argument("-bs", "--batch_size", type=int, help="the number of (default 14-day) windows to fit in each batch; default to running everything in one batch")
+    parser.add_argument("-is", "--increment_size", type=int, help="the number of windows to shift forward for each subsequent fit; default to 1")
+    parser.add_argument("-ws", "--window_size", type=int, help="the number of days in each TC-window; default to 14")
+    parser.add_argument("-tmn", "--tc_min", type=float, default=0, help="The lowest tc to allow")
+    parser.add_argument("-tmx", "--tc_max", type=float, default=0.99, help="The lowest tc to allow")
+    parser.add_argument("-rp", "--region_params", type=str, default="input/region_params.json", help="the path to the region-specific params file to use for fitting; default to 'input/region_params.json'")
+    parser.add_argument("-rg", "--regions", nargs="+", choices=all_regions.keys(), required=False, help="Specify the regions to be run, default is all regions (not counties)")
+    parser.add_argument("-wb", "--write_batch_output", action="store_true", default=False, help="write the output of each batch to the database")
+    parser.add_argument("-plt", "--plot", action="store_true", default=False, help="open a plot when fitting is complete, comparing modeled hosp to actual; default False")
+    fit_params = parser.parse_args()
+    look_back = fit_params.look_back
+    batch_size = fit_params.batch_size if fit_params.batch_size is not None else look_back
+    increment_size = fit_params.increment_size if fit_params.increment_size is not None else 1
+    window_size = fit_params.window_size if fit_params.window_size is not None else 14
+    regions = fit_params.regions
+    region_params = fit_params.region_params
+    write_batch_output = fit_params.write_batch_output
+
+    engine = db_engine()
+
+    # run fit
+    for region in regions:
+        print(f'Region: {all_regions[region]}')
+        try:
+            # get hospitalization data and adjust start/end dates
+            print("Adjusting start / end dates")
+            region_counties = json.load(open(region_params))[region]['county_names']
+            region_county_ids = json.load(open(region_params))[region]['county_fips']
+            hosps = ExternalHosps(db_engine()).fetch_from_db(region_county_ids)
+            fit_params.start_date = max(dt.datetime.strptime(fit_params.start_date, "%Y-%m-%d").date(),
+                                        min(hosps.index[hosps['currently_hospitalized'] > 0]) - dt.timedelta(days=1)).strftime("%Y-%m-%d")
+            fit_params.end_date = min(dt.datetime.strptime(fit_params.end_date, "%Y-%m-%d").date(),
+                                        max(hosps.index[hosps['currently_hospitalized'] > 0])).strftime("%Y-%m-%d")
```

Comment:
Can we assign different local variables "start_date" and "end_date" here, instead of "fit_params.start_date" and fit_params.end_date"? It's bugging me that we're resetting the parameters from the command line arguments...unless I'm misreading.